### PR TITLE
(#1035) Do case-insensitive fact value comparison

### PR DESCRIPTION
**Pull request by smortex**
Align with the Puppet way of comparing strings, that is ignore the case for comparison.

I see no test for this particular area of the code, but since I am probably not doing the right thing to run them I tested on my laptop by replacing the binary and it seems to be good.  Can't I test with:

```sh-session
romain@zappy ~/Projects/go-choria % go test
?   	github.com/choria-io/go-choria	[no test files]
romain@zappy ~/Projects/go-choria % echo $?
0
```

### CODE REVIEW

**Comment by smortex** [comment]
Similarly, it's possible to `include profile::base::${facts.get('os.family')}` in a puppet manifest, and the included class is `profile::base::debian` when `os.family=Debian` (switched to downcase).

Do you want me to adjust the class filter accordingly?

**Comment by ripienaar** [comment]
Yeah that would be good

**Comment by smortex** [comment]
:ballot_box_with_check: Done!

**Comment by smortex** [comment]
Oh, I was not sure about how to handle `reMatch`.  It looks like go does not have a flag to do a case insensitive matching of regexp, and I am not confident enough to add `(?i)` at the beginning of the value…  What do you think about this?

**Comment by ripienaar** [comment]
To test you can either do `go test ./...` or `ginkgo -r .`

File: filter/classes/classes.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ func hasClassMatching(needle string, stack []string) bool {
 
 func hasClass(needle string, stack []string) bool {
 	for _, class := range stack {
-		if class == needle {
+		if class == strings.ToLower(needle) {
```
Comment (ripienaar) [inline filter/classes/classes.go:66]: maybe do `lneedle := strings.ToLower(needle)` above the for loop then use `if class == lneedle`

**Review by ripienaar — CHANGES_REQUESTED** [review]
looks great, one tiny improvement to be quicker and more efficient

**Comment by ripienaar** [comment]
> Oh, I was not sure about how to handle `reMatch`. It looks like go does not have a flag to do a case insensitive matching of regexp, and I am not confident enough to add `(?i)` at the beginning of the value… What do you think about this?

looks like prepending that `(?i)` is about the only way it can be done

File: filter/classes/classes.go (hunk 1 above)
Comment (smortex) [reply filter/classes/classes.go]: Done!

**Comment by smortex** [comment]
Commits adjusted and rebased, thanks!

**Review by ripienaar — APPROVED** [review]

**Review by ripienaar — CHANGES_REQUESTED** [review]
We should add some tests I think, should be easy on the existing tests

**Review by ripienaar — APPROVED** [review]

**Pull request merged**